google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiDetailsPanel.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.apis;
+
+import com.google.cloud.tools.intellij.ui.BrowserOpeningHyperLinkListener;
+import com.google.cloud.tools.intellij.ui.GoogleCloudToolsIcons;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.google.cloud.tools.intellij.util.ThreadUtil;
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.common.collect.ImmutableList;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.util.SVGLoader;
+import java.awt.Image;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import javax.swing.ImageIcon;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextPane;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import org.jetbrains.annotations.Nullable;
+
+/** The form-bound class for the Cloud API details panel. */
+public final class GoogleCloudApiDetailsPanel {
+
+  private static final String JAVA_CLIENT_LANGUAGE = "java";
+  private static final String LINKS_SEPARATOR = " | ";
+
+  private JLabel icon;
+  private JLabel nameLabel;
+  private JLabel versionLabel;
+  private JLabel statusLabel;
+  private JPanel panel;
+  private JTextPane descriptionTextPane;
+  private JTextPane linksTextPane;
+
+  private CloudLibrary currentCloudLibrary;
+
+  /** Returns the {@link JPanel} that holds the UI elements in this panel. */
+  JPanel getPanel() {
+    return panel;
+  }
+
+  /**
+   * Sets the currently displayed {@link CloudLibrary} to the given one.
+   *
+   * <p>If the given {@link CloudLibrary} is already displayed, nothing is done and no UI updates
+   * are made.
+   *
+   * @param library the {@link CloudLibrary} to display
+   */
+  void setCloudLibrary(CloudLibrary library) {
+    if (cloudLibrariesEqual(currentCloudLibrary, library)) {
+      return;
+    }
+    currentCloudLibrary = library;
```

Comment:
library can't realistically be null, right?

Reply:
Right, if it's null, there's a bug somewhere so it should probably blow-up like this will.